### PR TITLE
Improved Template

### PR DESCRIPTION
**Pull request by jxstxn1**
Re-Improved our template
- Added `analysis_options.yaml` for `server/`
- Added `pubspec.yaml` to template to mananage template deps
- Tightened `analysis_options.yaml` in template folder

### CODE REVIEW

**Comment by jxstxn1** [comment]
@passsy is this approach better than adding the deps to `dev_dependencies`?
And any more feedback from you?

**Comment by passsy** [comment]
`pubspec.yml` still contains the dev_dependencies 🤔

**Comment by jxstxn1** [comment]
thanks, forgot to push

File: template/pubspec.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+environment:
+  sdk: ">=2.18.0 <3.0.0"
+
+dependencies:
+  dcli: ^1.36.0
+  shelf: ^1.4.0
+  shelf_helmet: ^1.1.0
+  shelf_router: ^1.1.3
+  shelf_static: ^1.1.0
+  sidekick_core: ^0.13.0
```
Comment (passsy) [inline template/pubspec.yaml:16]: You could add `dockerize_sidekick_plugin` as path dependency

**Review by robiness — APPROVED** [review]

**Pull request merged**